cmd/docgenerate/main.go

```diff
@@ -173,3 +199,70 @@ func versionPaths(srcdir string) []string {
 
 	return paths
 }
+
+func createIndexFile(parentPath string, srcPath string, version string) string {
+
+	type Index struct {
+		URL     string
+		Body    string
+		Title   string
+		Snippet string
+	}
+
+	files, err := ioutil.ReadDir(parentPath)
+	if err != nil {
+		log.Fatalln("ReadDir error:", err)
+	}
+
+	var contents []string
+
+	for _, f := range files {
+		n := f.Name()
+
+		if strings.HasPrefix(n, ".") {
+			continue
+		}
+
+		if f.IsDir() {
+			if n == "enterprise" {
+				continue
+			}
+			contents = append(contents, createIndexFile(path.Join(parentPath, n), srcPath, version))
+		} else {
+			ext := filepath.Ext(n)
+			if ext == ".md" {
+				tempPath := path.Join(parentPath, n)
+				tempFile, err := ioutil.ReadFile(tempPath)
+				if err != nil {
+					panic(err)
+				}
+				tempString := string(tempFile)
```

Comment:
This is a nit: it could make sense to move the parsing logic into its own function that takes a byte slice as input, and returns a byte slice to be appended to the index contents. This would be lines 239 to 261, essentially everything after reading the file bytes, up to and including the json.Marshal.